src/wallet/rpcwallet.cpp

```diff
@@ -3082,24 +3082,22 @@ UniValue bumpfee(const JSONRPCRequest& request)
             HelpExampleCli("bumpfee", "<txid>"));
     }
 
-    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VOBJ});
+    RPCTypeCheckObj(request.params, {
```

Comment:
IMO `RPCTypeCheckObj` is not very elegant. Here argument type is checked, and below it is parsed (which can fail) and sometimes  it is validated (for instance out of bounds). Not to mention the "argument is not required and can be null" cases.

Something would be cleaner (don't mind the API):
```cpp
uint256 hash = RequestParamUint256(request, "txid");
```
And if it has a default argument then it is not required:
```cpp
bool replaceable = RequestParamBool(request, "options.replaceable", true);
```
So the idea is that these `RequestParam*` calls check, parse and validate the argument.

This is obviously work for other PR, but if you drop the options *magic transformation* and `UniValueType` changes then there is no change here too.

Reply:
How would we detect unexpected params?